### PR TITLE
[FIX] survey: send email toggle button fix

### PR DESCRIPTION
**Pull request by ampa-odoo**
**Current behavior before PR:**
When user tried to share the survey and the access mode was public (Anyone with the link) , the 'Send by email' button was flickering and not working properly.

**Desired behavior after PR is merged:**
'Send by Email' button does not flicker and works properly.

**Task**-3240813

### CODE REVIEW

File: addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/** @odoo-module **/
+import { registry } from "@web/core/registry"
+import { BooleanToggleField } from '@web/views/fields/boolean_toggle/boolean_toggle_field';
+
+export class SurveyBooleanToggleField extends BooleanToggleField {
+    /**
+    *override
```
Comment (tde-banana-odoo) [inline addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field.js:7]: Would improve the docstring explaining the purpose of the override ^^

File: addons/survey/wizard/survey_invite.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def _get_default_author(self):
         ondelete='set null', default=_get_default_author)
     # recipients
     partner_ids = fields.Many2many(
-        'res.partner', 'survey_invite_partner_ids', 'invite_id', 'partner_id', string='Recipients',
+        'res.partner', 'survey_invite_partner_ids', 'invite_id', 'partner_id', string='Recipients', required=True,
```
[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/survey/wizard/survey_invite.py:42]: Can be added in the view also, to be more stable friendly.

File: addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/** @odoo-module **/
+import { registry } from "@web/core/registry"
+import { BooleanToggleField } from '@web/views/fields/boolean_toggle/boolean_toggle_field';
+
+export class SurveyBooleanToggleField extends BooleanToggleField {
```
Comment (tde-banana-odoo) [inline addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field.js:5]: Commit message does not explain "why" we have to do this... should contain the technical explanation as it is not obvious why we have to do this :)

File: addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    /**
+    *override
+    */
+    onChange(newValue) {
+        this.props.update(newValue);
```
Comment (flch-odoo) [inline addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field.js:10]: I'm genuinely unaware of this, is this good practice vs. updating the `props.record` like in the original component?

File: addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field.js (hunk 3 above)
Comment (flch-odoo) [inline addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field.js:5]: Maybe something specifying that we don't save could help here as well.
```suggestion
export class SurveyBooleanToggleFieldNoSaving extends BooleanToggleField {
```

File: addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field_no_saving.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+/** @odoo-module **/
+import { registry } from "@web/core/registry"
+import { BooleanToggleField } from '@web/views/fields/boolean_toggle/boolean_toggle_field';
+
+export class SurveyBooleanToggleFieldNoSaving extends BooleanToggleField {
```
Comment (tde-banana-odoo) [inline addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field_no_saving.js:5]: Have to say it is a bit verbose ^^

File: addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field.js (hunk 3 above)
Comment (ampa-odoo) [reply addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field.js]: Hey, @flch-odoo 
Done with all suggested changes!! Can you please check it out? 
Thanks:)

File: addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field_no_saving.js (hunk 5 above)
Comment (flch-odoo) [reply addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field_no_saving.js:5]: @ampa-odoo , in that case I would advise to drop the `Boolean` which doesn't add much .

Comment (ampa-odoo) [reply addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field_no_saving.js]: @flch-odoo Done with the changes. Please check it out!!

File: addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field.js (hunk 4 above)
Comment (flch-odoo) [reply addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field.js]: Sorry I have to insist here. Looking for why we don't see more like this, I found this PR which actively removed `props.update` calls: #112792.

Comment (ampa-odoo) [reply addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field.js]: Please check out the changes!!

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**